src/main/java/com/hivemq/util/ObjectMemoryEstimation.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019 dc-square GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hivemq.util;
+
+import com.hivemq.extension.sdk.api.annotations.Nullable;
+
+/**
+ * @author Lukas Brandl
+ */
+public class ObjectMemoryEstimation {
```

Comment:
Why not use the sizes from this "classic" article https://www.javaworld.com/article/2077408/sizeof-for-java.html?

Reply:
I'll go with the suggestions of Silvio. They seem to align with the article.